narwhal/storage/src/payload_store.rs

```diff
@@ -41,8 +42,12 @@ impl PayloadStore {
     }
 
     pub fn write(&self, digest: &BatchDigest, worker_id: &WorkerId) -> Result<(), TypedStoreError> {
+        fail_point!("narwhal-store-before-write");
+
         self.store.insert(&(*digest, *worker_id), &0u8)?;
         self.notify_subscribers.notify(&(*digest, *worker_id), &());
+
+        fail_point!("narwhal-store-after-write");
```

Comment:
I think it would be interesting to see the failure before `self.notify_subscribers.notify` as well